### PR TITLE
SDK3.0 Release

### PR DESCRIPTION
**Pull request by jimypbr**
# What does this PR do?

This makes Optimum compatible with the Poplar 3.0 SDK release. 
The biggest change is that we are now using poptorch's dispatch tracer, which replaces the JIT tracer.

The main benefits of this are:
1. Better type handling - we can get rid of `poptorch.autocast`
2. Native argument and output handling. We can pass kwargs and dictionarys to the IPU models and also use dictionary outputs. 
3. Faster tracing

### CODE REVIEW

File: optimum/graphcore/models/t5/modeling_t5.py [hunk 1]
```diff
@@ -286,23 +395,23 @@ def _forward(
...
-        if self.model_parallel:
-            torch.cuda.set_device(self.decoder.first_device)
-            hidden_states = hidden_states.to(self.decoder.first_device)
-            if decoder_input_ids is not None:
-                decoder_input_ids = decoder_input_ids.to(self.decoder.first_device)
-            if attention_mask is not None:
-                attention_mask = attention_mask.to(self.decoder.first_device)
-            if decoder_attention_mask is not None:
-                decoder_attention_mask = decoder_attention_mask.to(self.decoder.first_device)
+        # if self.model_parallel:
```
Comment (michaelbenayoun) [inline optimum/graphcore/models/t5/modeling_t5.py:406]: Should remove?

**Review by michaelbenayoun — APPROVED** [review]
LGTM, huge work @jimypbr !!!

File: optimum/graphcore/models/convnext/modeling_convnext.py [hunk 2]
```diff
@@ -24,6 +27,36 @@
...
+            x = x.float()
+            u = x.mean(1, keepdim=True)
+            s = (x - u).pow(2).mean(1, keepdim=True)
+            x = (x - u) / torch.sqrt(s + self.eps)
+            x = x.to(dtype=input_dtype)
+            x = self.weight[:, None, None] * x + self.bias[:, None, None]
+        return x
+
+
 @register(transformers.ConvNextForImageClassification)
```
Comment (gejinchen) [inline optimum/graphcore/models/convnext/modeling_convnext.py:60]: Maybe remove `transformers.`, just to be consistent with others.

File: examples/speech-recognition/run_speech_recognition_ctc.py [hunk 3]
```diff
@@ -740,7 +765,10 @@ def compute_metrics(pred):
         "finetuned_from": model_args.model_name_or_path,
         "tasks": "speech-recognition",
         "tags": ["automatic-speech-recognition", data_args.dataset_name],
-        "dataset_args": f"Config: {config_name}, Training split: {data_args.train_split_name}, Eval split: {data_args.eval_split_name}",
+        "dataset_args": (
```
[the remark this answers is not in the document]
Comment (jimypbr) [reply examples/speech-recognition/run_speech_recognition_ctc.py:768]: This is just copied from 4.20 version of this script.

File: optimum/graphcore/ipu_configuration.py [hunk 4]
```diff
@@ -221,6 +221,8 @@ def _to_options(self, for_inference: bool = False, compile_only: bool = False) -
         if self.enable_half_partials:
             opts.Precision.setPartialsType(torch.float16)
 
+        opts.Jit.traceModel(False)
```
[the remark this answers is not in the document]
Comment (jimypbr) [reply optimum/graphcore/ipu_configuration.py]: I don't need this anymore actually thanks.

File: optimum/graphcore/models/t5/modeling_t5.py (hunk 1 above)
Comment (jimypbr) [reply optimum/graphcore/models/t5/modeling_t5.py]: yes

File: optimum/graphcore/models/convnext/modeling_convnext.py (hunk 2 above)
Comment (jimypbr) [reply optimum/graphcore/models/convnext/modeling_convnext.py:60]: thanks

**Pull request merged**